docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java

```diff
@@ -75,15 +75,15 @@ public void testCommandIsExecuted() {
         @Rule
         // withMinimumDurationStrategy {
         public GenericContainer<?> bboxWithMinimumDuration = new GenericContainer<>(DockerImageName.parse("busybox:1.31.1"))
-            .withCommand("sh", "-c", String.format("sleep 5 && echo \"%s\"", HELLO_TESTCONTAINERS))
+            .withCommand("sh", "-c", String.format("sleep 1 && echo \"%s\" && sleep 5", HELLO_TESTCONTAINERS))
```

Comment:
I am not sure about this one. `MinimumDurationRunningStartupCheckStrategy` shouldn't rely on stdout, which makes me think that there is either a bug in TC, bug in Docker or some other bizarre stuff going on.

At least I cannot say that the previous command was wrong by the look of it

Reply:
In the test, we first wait for the `hello` to appear:
https://github.com/testcontainers/testcontainers-java/blob/bda75f796d9ae461f0e05169409f257c7a3e8720/docs/examples/junit4/generic/src/test/java/org/testcontainers/containers/startupcheck/StartupCheckStrategyTest.java#L32

This has nothing to do with `MinimumDurationRunningStartupCheckStrategy`, but with the way the test was written.

What would happen with the old code was sleeping for 5s, then `hello` gets printed, then the container exits. Afterwards, the test will inspect a terminated container, thus the test failing.

Now one might argue, why there is a need for `waitForHello(bboxWithMinimumDuration);` and printing to stdout in the first place in `minimumDurationStrategyTest`.

Reply:
ah, sorry, I missed that we actually wait for the logs.

Good point at why we need it at all 👍 Could as well be a separate test if needed

Reply:
I think this was purely a misunderstanding of how `MinimumDurationRunningStartupCheckStrategy` work at the time of writing the tests. 

I just minimized the test accordingly (although it doesn't still make much sense like this, without checking how long `MinimumDurationRunningStartupCheckStrategy` waited).